ls1intum/Orion
# [Feature] Download Submissions for Tutors

### Motivation and Context
This is the first step towards enabling assessment in Orion. This PR enables tutors to open an exercise for assessment as well as to download and test submissions locally. Actually adding/editing review comments will be adressed at a later time.

### Description
Orion can now handle tutor exercises as third exercise type. Upon entering the assessment dashboard of an exercise with Orion a button is presented to open that exercise as tutor/for assessment. This creates a new project with the test repository. When entering the assessment dashboard again users get presented with buttons to assess submissions in Orion. Clicking on those will download the submission into the correct subfolder. The submission can then be tested locally, a corresponding run configuration is already configured. It is also possible to edit and debug the submission with all available IDE tools. Further submissions can also be downloaded, replacing the current submission.

Currently the normal assessment editor is opened when assessing a submission, this will be changed in a later PR.

### Steps for Testing
Needs to be tested together with https://github.com/ls1intum/Artemis/pull/3530

1. Install the release as described in [the readme](https://github.com/ls1intum/Orion/blob/master/README.md#testing-of-pull-requests)
2. Change Settings -> Tools -> Orion -> Artemis URL to whatever testserver you are using
3. Deploy the mentioned Artemis PR
4. Create a java programming exercises with at least 2 participations
5. Log in as tutor in Orion
6. Navigate to the exercise's assessment dashboard
7. Start the assessment in Orion
     - It should open the exercise in a new project
     - The correct tests should be downloaded
     - The assessment dashboard should be opened by default
8. Start a new assessment
     - The submission should be downloaded
     - The submission's assessment editor should be opened by default (again, this will be changed later)
9. Run the submission locally
     - A run configuration should be available
10. Edit the submission/the tests
     - Everything (syntax highlighting, autocompletion, ...) should work normally
11. Go back to the exercise dashboard and start a new assessment
     - The current submission should be replaced
     - Everything else should still work
12. It should be possible to repeatedly change submissions through the dashboard

#### Additional Check
1. Cloning in general was refactored, try to clone any exercise as student or instructor and make sure this still works correctly

### Screenshots
The buttons should be easy to find, if any button is too hidden please report it; the actual users will not have any guidance
